packages/components/src/mobile/bottom-sheet/range-cell.native.js

```diff
@@ -1,46 +1,180 @@
+		this.onCellPress = this.onCellPress.bind( this );
+
+		const initialValue = this.validateInput( props.value || props.defaultValue || props.minimumValue );
+
+		this.state = { accessible: true, sliderValue: initialValue, initialValue, hasFocus: false };
+	}
+
+	componentDidUpdate( ) {
+		const reset = this.props.value === null;
+		if ( reset ) {
+			this.handleReset();
+		}
+	}
+
+	componentWillUnmount() {
+		this.handleToggleFocus();
+	}
+
+	handleChange( text ) {
+		if ( ! isNaN( Number( text ) ) ) {
+			this.setState( { sliderValue: text } );
+			this.announceCurrentValue( text );
+		}
+	}
+
+	handleReset() {
+		this.handleValueSave( this.props.defaultValue || this.state.initialValue );
+	}
+
+	handleToggleFocus( validateInput = true ) {
+		const newState = { hasFocus: ! this.state.hasFocus };
+
+		if ( validateInput ) {
+			const sliderValue = this.validateInput( this.state.sliderValue );
+			this.handleValueSave( sliderValue );
+		}
+
+		this.setState( newState );
+	}
+
+	validateInput( text ) {
+		const { minimumValue, maximumValue } = this.props;
+		if ( ! text ) {
+			return minimumValue;
+		}
+		if ( typeof text === 'number' ) {
+			return Math.min( Math.max( text, minimumValue ), maximumValue );
+		}
+		return Math.min( Math.max( text.replace( /[^0-9]/g, '' ).replace( /^0+(?=\d)/, '' ), minimumValue ), maximumValue );
+	}
+
+	handleValueSave( text ) {
+		if ( ! isNaN( Number( text ) ) ) {
+			this.onChangeValue( text );
+			this.setState( { sliderValue: text } );
+			this.announceCurrentValue( text );
+		}
+	}
+
+	onChangeValue( initialValue ) {
+		const { minimumValue, maximumValue, setAttributes, attribute } = this.props;
+
+		let sliderValue = initialValue;
+		if ( sliderValue < minimumValue ) {
+			sliderValue = minimumValue;
+		} else if ( sliderValue > maximumValue ) {
+			sliderValue = maximumValue;
+		}
+		setAttributes( {
+			[ attribute ]: sliderValue,
+		} );
+	}
+
+	onCellPress() {
+		this.setState( { accessible: false } );
+		if ( this.sliderRef ) {
+			const reactTag = findNodeHandle( this.sliderRef );
+			AccessibilityInfo.setAccessibilityFocus( reactTag );
+		}
+	}
+
+	announceCurrentValue( value ) {
+		const announcement = sprintf( __( 'Current value is %s' ), value );
+		AccessibilityInfo.announceForAccessibility( announcement );
+	}
+
+	render() {
+		const {
+			value,
+			defaultValue,
+			minimumValue = 0,
+			maximumValue = 10,
+			disabled,
+			step = 1,
+			minimumTrackTintColor = '#00669b',
+			maximumTrackTintColor = Platform.OS === 'ios' ? '#e9eff3' : '#909090',
+			thumbTintColor = Platform.OS === 'android' && '#00669b',
+			...cellProps
+		} = this.props;
+
+		const { hasFocus, sliderValue, accessible } = this.state;
+
+		const accessibilityLabel =
+		sprintf(
+			/* translators: accessibility text. Inform about current value. %2$s: Current value. */
+			_x( '%1$s. Current value is %2$s', 'Slider for picking a number inside a range' ),
+			cellProps.label, value
+		);
```

Comment:
It would probably be good to add what `%1$s` is for.
This text is the guide and (usually) only context for translators to do a good translation work. The better context, the better translations we get :)

Something like: `%1$s: Control label` or similar should help 👍

Reply:
Added! Thanks!